js_modules/dagster-ui/packages/ui-core/src/runs/filterLogs.tsx

```diff
@@ -49,3 +51,23 @@ export function filterLogs(logs: LogsProviderLogs, filter: LogFilter, filterStep
     textMatchNodes,
   };
 }
+
+// Given an array of metadata entries, returns searchable text in the format:
+// [`label1:value1`, ...], where "value1" is the top-level value(s) present in the
+// metadata entry object converted to JSON. All of our metadata entry types contain
+// different top-level keys, such as "intValue", "mdStr" and "tableSchema", and
+// the searchable text is the value of these keys.
+//
```

Comment:
How big can the `metadataEntries` object be? Could this stringifying and searching become sluggish (or memory intensive, if we're constantly stringifying everything, e.g. on every keystroke)  for large metadata sets on runs with zillions of logs?

Reply:
Similar perf concern from me. I think we only need to stringify once for each message. We could accomplish that by using a weakmap to memoize `metadataEntryKeyValueStrings`.

Reply:
Hmmmm I like the weakmap idea, I hadn't thought of the evaluated-per-keystroke implications. Will update this!

Reply:
Wrapped this in our `weakmapMemoize` helper and used some console.logs in the body of the function to verify that it runs on the first keystroke and not on subsequent keystrokes 🙌  Good call!  I think that would have been bad...